### PR TITLE
[feat](mtmv)mtmv support paimon partition refresh

### PR DESCRIPTION
**Pull request by zddr**
### What problem does this PR solve?
Previously, when using Paimon to create MTMV, it was not possible to perceive changes in partition lists and data, so only `refresh materialized view mv1 complete` could be used to force full refresh.

This PR obtains the partition list of Paimon, the last update time of the partition, and the latest snapshotId of the table.

Therefore, MTMV can be partitioned based on Paimon tables and perceive changes in data, automatically refreshing partitions

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
mtmv support paimon partition refresh
### Release note

mtmv support paimon partition refresh

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [x] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run external

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonPartition.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// under the License.
+
+package org.apache.doris.datasource.paimon;
+
+public class PaimonPartition {
+    private String partitionValues;
+    private long recordCount;
+    private long fileSizeInBytes;
+    private long fileCount;
+    private long lastUpdateTime;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonPartition.java:25]: add comment to explain these attributes.

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonPartition.java:25]: make them final

File: fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonExternalTable.java [hunk 2]
```diff
@@ -73,18 +95,95 @@ public Table getPaimonTable() {
         return schemaCacheValue.map(value -> ((PaimonSchemaCacheValue) value).getPaimonTable()).orElse(null);
     }
 
+    public PaimonPartitionInfo getPartitionInfoFromCache() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonExternalTable.java:98]: private?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonExternalTable.java [hunk 3]
```diff
@@ -73,18 +95,95 @@ public Table getPaimonTable() {
...
+        makeSureInitialized();
+        Optional<SchemaCacheValue> schemaCacheValue = getSchemaCacheValue();
+        if (!schemaCacheValue.isPresent()) {
+            return new PaimonPartitionInfo();
+        }
+        return ((PaimonSchemaCacheValue) schemaCacheValue.get()).getPartitionInfo();
+    }
+
+    public List<Column> getPartitionColumnsFromCache() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/PaimonExternalTable.java:107]: private?

**Comment by zddr** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by seawinde — APPROVED** [review]

**Pull request merged**